### DISCUSSION TITLE
+page.js Boilerplate

### DISCUSSION THREAD

**Discussion by pevey** [sveltejs/kit#7011] [Q&A] [answered]
In the app I'm working on, I've ended up with a lot of +page.js files, one in almost every route, that simply does this:

```
export async function load({ data }) {
    return {
        ...data
    }
}
```

The point is just to pass along the exports from +page.server.js to +page.svelte.  Am I approaching this wrong and missing a better way to structure this?  It's so unlike Svelte and SvelteKit to have any unnecessary boilerplate, so I feel like I must be missing something.  

I feel like data returned from the +page.server.js load function should be available in the +page.svelte route without the extra +page.js file in every route.

**Comment by cdcarson** [comment, marked as answer]
You don't need `+page.js` at all. You'll get the data directly from `+page.server.js`

**Reply by pevey** [reply]
Oh, wow.  I just tried it, and you are correct.  I've seen this pattern in so many examples and migration stories that I just blindly followed and didn't try without it.

Also, in the docs (https://kit.svelte.dev/docs/load), it says:

`In other words +page.server.js passes data along to +page.js, which passes data along to +page.svelte.`

It's a bit confusing.  In any case, thank you!

**Reply by cdcarson** [reply]
Yeah, the section definitely needs some info about how data from the various  `load`s are merged.
